tests/recipe/CREATE_recipes.test.js

```diff
@@ -0,0 +1,130 @@
+const supertest = require("supertest");
+const app = require("../../app");
+const Recipe = require("../../models/recipe");
+const User = require("../../models/user");
+const api = supertest(app);
+const mongoose = require("mongoose");
+
+const {
+  initialUsers,
+  recipesInDb,
+  recipeToAdd,
+  recipeReview,
+} = require("../test_helper");
+const STATUS_CODE = require("../../shared/errorCode");
+
+let authHeader;
+let userId;
+let recipeId;
+let token;
+
+describe("recipe api", () => {
+  beforeEach(async () => {
+    await User.deleteMany({});
+    const newUser = initialUsers[1];
+    let response = await api.post("/api/users").send(newUser);
+
+    response = await api
+      .post("/api/auth/login")
+      .send(newUser)
+      .expect(STATUS_CODE.OK);
+
+    token = response.body.token;
+    authHeader = `Bearer ${token}`;
+    userId = response.body.id;
+  });
+
+  describe("add new recipe", () => {
+    test("can be added", async () => {
+      await Recipe.deleteMany({});
+      const recipe = recipeToAdd;
+      const beforeRecipes = await recipesInDb();
+
+      const response = await api
+        .post("/api/recipe/add")
+        .set("Authorization", authHeader)
+        .send(recipe)
+        .expect(STATUS_CODE.CREATED)
+        .expect("Content-Type", /application\/json/);
+
+      recipeId = response.body._id;
+
+      const afterRecipes = await recipesInDb();
+      expect(afterRecipes).toHaveLength(beforeRecipes.length + 1);
```

Comment:
```suggestion
     let recipes = await recipesInDb();
      expect(recipes).toHaveLength(0);
      
      const response = await api
        .post("/api/recipe/add")
        .set("Authorization", authHeader)
        .send(recipe)
        .expect(STATUS_CODE.CREATED)
        .expect("Content-Type", /application\/json/);
      
      recipes = await recipesInDb();
      expect(recipes).toHaveLength(1);
```

You should make the test as simple as possible. You shouldn't make the reader have to think about beforeRecipes and afterRecipes, etc.

You should probably assert that the added recipes is the correct one too.